app/assets/javascripts/views/abstract/pure_view_ctrl.ts

```diff
@@ -63,9 +63,15 @@ export class PureViewCtrl<P = CtrlProps, S = CtrlState> {
     if (!this.$timeout) {
       return;
     }
-    this.state = Object.freeze(Object.assign({}, this.state, state));
     return new Promise((resolve) => {
-      this.stateTimeout = this.$timeout(resolve);
+      this.stateTimeout = this.$timeout(() => {
+        /**
+         * State changes must be *inside* the timeout block for them to be affected in the UI
+         * Otherwise UI controllers will need to use $timeout everywhere
+         */
+        this.state = Object.freeze(Object.assign({}, this.state, state));
+        resolve();
+      });
```

Comment:
Wait what? Was this an issue you encountered during the development of this feature? Because otherwise changing state outside the timeout block means we don't have to await a whole angular digest cycle when we're just manipulating values.

Reply:
Actually please address this one before merging. It may break other parts of the app.

Reply:
So the history of this function was:
- it first performed the state change in the timeout 
- after I handed it off to you and you got familiar with it you thought it would make sense to put it outside the timeout. I was hands-off at that point and couldn't really test it but it seemed like a positive change
- after building my feature, it took me back to the original design purpose: setState is designed to trigger a UI refresh, the same way it does in React via render(). If you call setState() in Angular from a callback, it won't update the UI, unless you do it in a $timeout block. For example, [this](https://github.com/standardnotes/web/blob/develop/app/assets/javascripts/views/application_group/application_group_view.ts#L20) wouldn't work if not wrapped in $timeout. Or perhaps something simpler like [this](https://github.com/standardnotes/web/blob/develop/app/assets/javascripts/views/footer/footer_view.ts#L113) wouldn't work either. If it does work, it's just timing, since the `checkForSecurityUpdate` is local and will return quickly. But you can test this by creating an async function that takes 1s to return, and doing a state change in a `then`. It won't work unless you put the state change in a $timeout.

So if you want a setState for a purpose other than updating the UI, we should probably create another function for it, `setStateImmediate` or something like that.

Reply:
This was the function I was looking for: https://github.com/standardnotes/web/blob/app-group/app/assets/javascripts/views/account_switcher/account_switcher.ts#L30 Wouldn't work unless state change happen in $timeout block.

Reply:
Okay. We'll have to keep an eye out then. This could break other unrelated code.